### ISSUE TITLE
dosbox-staging fails to build at final linking stage

### ISSUE THREAD

**Issue by Zaechus** [NixOS/nixpkgs#211676]
### Steps To Reproduce
1. build dosbox-staging

### Build log
https://gist.github.com/Zaechus/acae0ae3a86558e77ee824d7caa6b750

### Notify maintainers
@JoshuaFern @AndersonTorres 

### Metadata
```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.86, NixOS, 23.05 (Stoat), 23.05.20230111.6c8644f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(root): `"nixos-22.05"`
 - channels(zaechus): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by keenanweaver** [comment]
Staging recently added [sdl2_image to the build dependencies](https://github.com/keenanweaver/dosbox-staging/commit/083dfb93c5d76a7bddfeb7580069447325030f01).

**Comment by AndersonTorres** [comment]
Unscramble this will take a longer time than I expected. There are at least 10 subprojects to grab.

**Comment by Zaechus** [comment]
In the meantime I have a custom package of a newer dosbox-staging version that builds properly:

`pkgs/dosbox-staging/default.nix`:
```
{ stdenv
, fetchFromGitHub
, makeWrapper
, meson
, ninja
, pkg-config
, alsa-lib
, fluidsynth
, glib
, libmt32emu
, libogg
, libpng
, libslirp
, opusfile
, SDL2
, SDL2_image
, SDL2_net
, speexdsp
, my
}:

stdenv.mkDerivation rec {
  pname = "dosbox-staging";
  version = "0.80.1";

  src = fetchFromGitHub {
    owner = pname;
    repo = pname;
    rev = "v${version}";
    hash = "sha256-I90poBeLSq1c8PXyjrx7/UcbfqFNnnNiXfJdWhLPGMc=";
  };

  nativeBuildInputs = [
    makeWrapper
    meson
    ninja
    pkg-config
  ];

  buildInputs = [
    alsa-lib
    fluidsynth
    glib
    libmt32emu
    libogg
    libpng
    libslirp
    opusfile
    SDL2
    SDL2_image
    SDL2_net
    speexdsp
    my.iir1
  ];

  NIX_CFLAGS_COMPILE = [
    "-I${SDL2_image}/include/SDL2"
    "-I${SDL2_net}/include/SDL2"
  ];

  configurePhase = ''
    meson setup --wrap-mode=nodownload build/release
  '';

  enableParallelBuilding = true;

  buildPhase = ''
    meson compile -C build/release
  '';

  installPhase = ''
    meson install -C build/release --destdir $out
    mkdir -p $out/bin
    ln -s $out/usr/local/bin/dosbox $out/bin/dosbox
  '';

  postFixup = ''
    mv $out/bin/dosbox $out/bin/${pname}
    makeWrapper $out/bin/dosbox-staging $out/bin/dosbox
  '';
}
```
`pkgs/iir1/default.nix`:
```
{ stdenv
, fetchFromGitHub
, cmake
}:

stdenv.mkDerivation rec {
  pname = "iir1";
  version = "1.9.3";

  src = fetchFromGitHub {
    owner = "berndporr";
    repo = pname;
    rev = "${version}";
    hash = "sha256-YC5QbuhU0rB3SydJ6wDkK3O+eSYdK1+3nIAcrQhAcHE=";
  };

  nativeBuildInputs = [
    cmake
  ];

  buildPhase = ''
    cmake .
    make
  '';

  # installPhase = ''
  #   make install
  # '';
}
```

**Comment by Zaechus** [comment]
This should be fixed by #215686.
